### DISCUSSION TITLE
add-launch-to-msi.js fails

### DISCUSSION THREAD

**Discussion by hrzafer** [wiverson/maven-jpackage-template#45] [Q&A]
For some reason, the script fails for my project but works on the template project. Her is the error I see in `msi-result.log`.

```
add-launch-to-msi.js(11, 1) Microsoft JScript runtime error: 'view.Fetch()' is null or not an object
```

I compared everything I can btw the two projects but couldn't find any difference which might cause this problem. 

From the error message, I can see that the following query returns null:

```
"SELECT File FROM File WHERE FileName='${app.name}.exe'"
```

**Comment by wiverson** [comment]
Looks like the app name (set in the Maven config) isn't matching the .exe name that the JavaScript is expecting. Could try just hardcoding in the name of the executable instead of relying on the Maven property.

BTW, I have no idea what happens if you have spaces in the exe name and or the path. Lots of tools get confused by spaces in the path.

**Reply by hrzafer** [reply]
Looking at the `add-launch-to-msi.js` under `target`, I can see that exe file name was injected into the SQL properly and there is no spaces in the file name.  After installation, I can manually start the application which indicates that there really is an MyApp.exe in the msi.

**Reply by wiverson** [reply]
Huh. Nothing comes to mind here. 🤷‍♂️

**Reply by wiverson** [reply]
If the template works but your app doesn't, if it was me I'd do a diff between the projects and line by line check/compare/try to isolate the difference between them. I'm sorry, can only help you do that as consulting engagement.

**Comment by wiverson** [comment]
Another user just contributed a fix that might help your situation, haven't rolled it into the template but will after the holidays.

https://github.com/wiverson/maven-jpackage-template/issues/46
